src/Controller/Controller.php

```diff
@@ -155,31 +151,6 @@ class Controller implements EventListenerInterface, EventDispatcherInterface
      */
     protected $_components;
 
-    /**
-     * Array containing the names of components this controller uses. Component names
-     * should not contain the "Component" portion of the class name.
-     *
-     * Example:
-     * ```
-     * public $components = ['RequestHandler', 'Acl'];
-     * ```
-     *
-     * @var array
-     * @link https://book.cakephp.org/3.0/en/controllers/components.html
-     *
-     * @deprecated 3.0.0 You should configure components in your Controller::initialize() method.
-     */
-    public $components = [];
```

Comment:
For this and `$helpers` I wonder if we should trigger an error if we see them in use? We could add a `__set` that emits a warning telling people what they should be doing instead? Right now post upgrade their app would not work and figuring out why may take some time.